### ISSUE TITLE
理解接口

### ISSUE THREAD

**Issue by XinbeiCai** [nnnth/UniLIP#9]
请问可以提供一个理解的推理接口嘛？现在的generate函数似乎有点问题

**Comment by nnnth** [comment]
Sorry for the late reply. As we only retrain the ViT in InternVL3, you can use the [inference script of InternVL3](https://huggingface.co/OpenGVLab/InternVL3-1B#inference-with-transformers), then replace the ViT with the UniLIP weight. 
1. You need to use this [script](https://github.com/nnnth/UniLIP/blob/main/tokenizer/extract_vit.py) to convert our [ckpt](https://huggingface.co/kanashi6/UniLIP) to be compatible with InternVL3.
2. Then you can refer the code [here](https://github.com/nnnth/UniLIP/blob/299817d6d7466d1b350ed0eb894c69bf9298e8e8/tokenizer/inference.py#L128) to replace the ViT weight in InternVL3.

If you want to evaluate understanding performance, here is the [instruction](https://github.com/nnnth/UniLIP/blob/main/tokenizer/README_RECON.md#understanding-benchmarks).

**Comment by XinbeiCai** [comment]
Thanks a lot！
I have another question:

In **Appendix C.4,** you mention that you “treat the output query embeddings as text embeddings and decode them into text.”
Could you clarify:
**How exactly do you decode the query embeddings?**
- Do you feed them back into the LLM decoder?
 - Or just apply the lm_head to map them to vocab logits?
 Could you share **a minimal code** snippet for this decoding step?

**Comment by XinbeiCai** [comment]
thx a lot~~

**Comment by KingH12138** [comment]
> thx a lot~~

Hey friend, have you written the understanding script yet? Could I take a look at it for reference?

**Comment by KingH12138** [comment]
Alright, I've got it figured out. Thanks to the silent author.
